### PR TITLE
Groups API Changes - Get Users in a Group & Update users in a group

### PR DESCRIPTION
**Pull request by JoyalAJohney-KeyValue**

### CODE REVIEW

File: test/authorization/service/group.service.test.ts [hunk 1]
```diff
@@ -44,7 +46,11 @@ describe('test Group Service', () => {
   const groupCacheService = Substitute.for<GroupCacheService>();
   const redisCacheService = Substitute.for<RedisCacheService>();
   const groupRoleRepository = Substitute.for<Repository<GroupRole>>();
+  const userRepository = Substitute.for<Repository<User>>();
   const roleRepository = Substitute.for<Repository<Role>>();
+  const connectionMock = Substitute.for<Connection>();
+  const userCacheService = Substitute.for<UserCacheService>();
```
Comment (doylefermi-kv) [inline test/authorization/service/group.service.test.ts:52]: Can we add tests to cover this new change?

**Review by doylefermi-kv — CHANGES_REQUESTED** [review]

File: src/authorization/graphql/group.graphql [hunk 2]
```diff
@@ -41,6 +45,7 @@ type Mutation {
   createGroup(input: NewGroupInput!): Group
   updateGroup(id: ID!, input: UpdateGroupInput!): Group
   deleteGroup(id: ID!): Group
+  updateGroupUsers(id: ID!, input: UpdateGroupUserInput!): [User]
```
[the remark this answers is not in the document]
Comment (JoyalAJohney-KeyValue) [reply src/authorization/graphql/group.graphql:48]: Wouldn't it be better to separate the functionality. We already have separate function to update userGroups

[the remark this answers is not in the document]
Comment (JoyalAJohney-KeyValue) [reply src/authorization/graphql/group.graphql:48]: Noted

File: src/authorization/resolver/group.resolver.ts [hunk 3]
```diff
@@ -29,6 +30,13 @@ export class GroupResolver {
     return this.groupService.getGroupById(id);
   }
 
+  @ResolveField('users')
+  async getUsersInGroupResolveField(group: Group) {
```
Comment (doylefermi-kv) [inline src/authorization/resolver/group.resolver.ts:34]: It is understood that this is a Resolve Field of `Users` in the Parent `Group`

`getUsersByGroup` would suffice right?

File: src/authorization/service/group.service.ts [hunk 4]
```diff
@@ -146,6 +164,40 @@ export class GroupService {
     return permissions;
   }
 
+  async updateGroupUsers(id: string, userIds: string[]): Promise<User[]> {
+    await this.getGroupById(id);
+    const usersInRequest = await this.userRepository.findByIds(userIds);
+    const validUsersInRequest: Set<string> = new Set(
```
Comment (doylefermi-kv) [inline src/authorization/service/group.service.ts:170]: Can we move this validation logic to a private function: `validateUsers(User[])`?

File: src/authorization/resolver/group.resolver.ts (hunk 3 above)
Comment (JoyalAJohney-KeyValue) [reply src/authorization/resolver/group.resolver.ts:34]: Yes

File: src/authorization/service/group.service.ts (hunk 4 above)
Comment (JoyalAJohney-KeyValue) [reply src/authorization/service/group.service.ts:170]: usersInRequest and validUsersInRequest are used in the remaining part of code. Should we move it to another function in that case?

Comment (doylefermi-kv) [reply src/authorization/service/group.service.ts:170]: We will be needing just the valid users in the request right?

**Review by doylefermi-kv — CHANGES_REQUESTED** [review]

**Review by doylefermi-kv — APPROVED** [review]

**Pull request merged**